### PR TITLE
UUID v4

### PR DESCRIPTION
**Pull request by radar**
For #43 and #53. It seems like the way to validate it is as straight-forward as a regex -- according to online advice. 

I noticed that some of the test descriptions were copy-pasted and not updated, so I fixed those too.

### CODE REVIEW

File: spec/unit/predicates/uuid_v4_spec.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+require 'dry/logic/predicates'
```
[the remark this answers is not in the document]
Comment (radar) [reply spec/unit/predicates/uuid_v4_spec.rb]: This is seemingly not required / used on other tests. Sooooo shhh, bot.

File: lib/dry/logic/predicates.rb [hunk 2]
```diff
@@ -197,6 +197,13 @@ def case?(pattern, input)
           pattern === input
         end
 
+        def uuid_v4?(input)
+          return false unless str?(input)
```
Comment (solnic) [inline lib/dry/logic/predicates.rb]: This is not needed, predicates should check just one thing and assume the type is correct. In actual usage like `dry-schema` you'd have `required(:uuid).filled(:string, :uuid_v4?)`.

Comment (radar) [reply lib/dry/logic/predicates.rb]: Okay, I think that's fine. I'll update the PR to reflect that.

File: spec/unit/predicates/uuid_v4_spec.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (solnic) [reply spec/unit/predicates/uuid_v4_spec.rb]: Yeah this is nothing to worry about. Better to mass-fix this in master.

**Pull request merged**